compiler/rustc_middle/src/ty/diagnostics.rs

```diff
@@ -63,16 +68,40 @@ impl<'tcx> TyS<'tcx> {
 
     /// Whether the type can be safely suggested during error recovery.
     pub fn is_suggestable(&self) -> bool {
-        !matches!(
-            self.kind(),
+        fn generic_arg_is_suggestible(arg: GenericArg<'_>) -> bool {
+            match arg.unpack() {
+                GenericArgKind::Type(ty) => ty.is_suggestable(),
+                _ => true,
```

Comment:
it probably makes sense to also return false for `ConstKind::Infer | ConstKind::Bound | ConstKind::Placeholder | ConstKind::Error` here. And also look at array lengths now.

After that r=me unless @estebank or @cjgillot want some other changes here